week1/homework/src/server.js

```diff
@@ -1,20 +1,45 @@
 'use strict';
 
+const url = require('url');
 const http = require('http');
 
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
 function createServer(port) {
   let state = 10;
+  let newState = state;
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    const requestedUrl = url.parse(request.url, true);
+    switch (requestedUrl.pathname) {
+      case '/state':
+        writeRespond(200, { state: newState }, response);
+        break;
+      case '/add':
+        newState++;
+        writeRespond(200, { state: newState }, response);
+        break;
+      case '/subtract':
+        newState--;
+        writeRespond(200, { state: newState }, response);
+        break;
+      case '/reset':
+        newState = state;
+        writeRespond(200, { state: newState }, response);
+        break;
+      default:
+        writeRespond(404, { error: 'Not found' }, response);
+    }
   });
-
   return server;
 }
 
+function writeRespond(statuCode, stateMsg, response) {
```

Comment:
great that you're passing the `response` to the function. This way this function is totally autonomous and it's easier to test (you'll learn about unit tests in one of the future lessons)